src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -149,74 +85,77 @@ class WeekCalendar extends Component<WeekCalendarProps, State> {
         key={item}
         current={item}
         firstDay={firstDay}
-        style={this.getWeekStyle(this.containerWidth, style)}
+        style={getWeekStyle}
         markedDates={markedDates}
         onDayPress={onDayPress}
         context={currentContext}
         numberOfDays={context.numberOfDays}
         timelineLeftInset={context.timelineLeftInset}
       />
     );
-  };
+  },[]);
 
-  getItemLayout = (_: any, index: number) => {
-    return {
-      length: this.containerWidth,
-      offset: this.containerWidth * index,
-      index
-    };
-  };
-
-  keyExtractor = (_: string, index: number) => index.toString();
+  const keyExtractor = useCallback((_, index: number) => index.toString(), []);
 
-  renderWeekDaysNames = () => {
+  const renderWeekDaysNames = useMemo(() => {
     return (
-      <WeekDaysNames 
-        firstDay={this.props.firstDay} 
-        style={this.style.dayHeader} 
+      <WeekDaysNames
+        firstDay={firstDay}
+        style={dayHeader}
       />
     );
-  };
+  },[firstDay, dayHeader]);
 
-  render() {
-    const {allowShadow, firstDay, hideDayNames, current, context} = this.props;
-    const {items} = this.state;
-    const extraData = Map({
-      current,
-      date: context.date,
-      firstDay
-    });
+  const extraData = Map({
+    current,
+    date: context.date,
+    firstDay
+  });
 
-    return (
-      <View
-        testID={this.props.testID}
-        style={[allowShadow && this.style.containerShadow, !hideDayNames && this.style.containerWrapper]}
-      >
-        {!hideDayNames && (
-          <View style={[this.style.week, this.style.weekCalendar]}>
-            {/* {this.props.weekNumbers && <Text allowFontScaling={false} style={this.style.dayHeader}></Text>} */}
-            {this.renderWeekDaysNames()}
-          </View>
-        )}
-        <FlatList
-          ref={this.presenter.list}
-          data={items}
-          extraData={extraData}
-          style={this.style.container}
-          horizontal
-          showsHorizontalScrollIndicator={false}
-          pagingEnabled
-          scrollEnabled
-          renderItem={this.renderItem}
-          keyExtractor={this.keyExtractor}
-          initialScrollIndex={NUMBER_OF_PAGES}
-          getItemLayout={this.getItemLayout}
-          onScroll={this.onScroll}
-          onMomentumScrollEnd={this.onMomentumScrollEnd}
-        />
-      </View>
-    );
+  return (
+    <View
+      testID={props.testID}
+      style={[allowShadow && defaultStyle.containerShadow, !hideDayNames && defaultStyle.containerWrapper]}
+    >
+      {!hideDayNames && (
+        <View style={[defaultStyle.week, defaultStyle.weekCalendar]}>
+          {renderWeekDaysNames}
+        </View>
+      )}
+          <FlashList
+            ref={presenter.list as unknown as LegacyRef<FlashList<string>>}
+            data={items}
+            extraData={extraData}
```

Comment:
The `extraData` will cause a re-render whenever the context's `date` changes, but also the `onScroll` is setting the context date (setDate) and setting the state's `items`. It's a mess (legacy I know) but let's change it if we're refactoring here. We should only have a `useEffect` for the context's `date` prop that changes the `items`. All other sources of updates feel redundant. What do you think?